SQLQuery/src/main/java/com/oops/app/Controller.java

```diff
@@ -0,0 +1,219 @@
+package com.oops.app;
+
+import java.util.List;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.oops.app.exceptions.BadRequestException;
+import com.oops.app.requestType.SolverRequest;
+import com.oops.app.requestType.PrivilageRequest;
+import com.oops.app.responseType.Solver;
+import com.oops.app.responseType.TaskQueue;
+import com.oops.app.responseType.Privilage;
+import com.oops.app.responseType.Solution;
+import com.oops.app.responseType.User;
+
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+@SpringBootApplication
+@RestController
+public class Controller {
+    private static SQLController sqlController = new SQLController();
+    public static void main( String[] args ) {
+        SpringApplication.run(Controller.class, args);
+    }
+
+    @GetMapping("/users")
+    public List<User> allUsers() {
+        return sqlController.getAllUser();
+    }
+
+    @GetMapping("/users/{username}")
+    @ResponseBody
+    public ResponseEntity<User> getUserByName(@PathVariable String username) {
+        User user = sqlController.getUser(username);
+        if(user == null) {
+            
+        }
+        return new ResponseEntity<User>(user, HttpStatus.OK);
+    }
+
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "OK", content = @Content),
+        @ApiResponse(responseCode = "400", description = "BAD REQUEST", content = @Content)
+    })
+    @PostMapping("/users")
+    @ResponseBody
+    public ResponseEntity<User> addUser(@RequestBody User newUser) {
+        User user = sqlController.addUser(newUser);
+        if(user == null) {
+            throw new BadRequestException();
+        }
+        return new ResponseEntity<User>(HttpStatus.OK);
+    }
+
+    @DeleteMapping("/users/{username}")
+    @ResponseBody
+    public ResponseEntity<User> deleteUser(@PathVariable String username) {
+        int status = sqlController.deleteUser(username);
+
+        switch (status) {
+            case 200: return new ResponseEntity<>(HttpStatus.OK);
+            default: return new ResponseEntity<>(HttpStatus.I_AM_A_TEAPOT);
```

Comment:
Well the 418 error code is an April fools joke and is not meant to be implemented. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418 It should never be able to get down to the default case. Any other error code whould be hard to make sense since it should be handeld by another case and not the default.

Reply:
If it is never meant to reach the default case, what is the point of using the switch statement? :O